docs/modules/sql/pages/user-defined-types.adoc

```diff
@@ -64,30 +51,34 @@ To create a new type, use the `CREATE TYPE` statement:
-<3> `compact` format requires `compactTypeName` - this is not the name of the created type, but rather internal name of the Compact record type, used internally by the Compact Serialization format.
+<1> In `java` format, you can use `javaClass` option to override the type class with a subclass. Note that you must provide a concrete class for `INSERT` statement to work, which makes `javaClass` option mandatory when the original type class is abstract or interface.
+<2> In `portable` format, you can use `portableFactoryId`, `portableClassId` and `portableVersion` options to specify a portable ID for the type, which is only effective when the type is used for `__key` or `this` fields. `portableVersion` is optional, similarly to `CREATE MAPPING` statement. However, it is not possible to inherit some components from the parent, so portable ID components must be defined together. Incomplete definitions have the same effect as no definition.
+<3> In `compact` format, you can use `compactTypeName` option to specify the Compact record name. If unspecified, it is defaulted to `<TypeName>CompactType`.
+<4> In `avro` format, you can use `avroSchema` option to specify an inline Avro schema, which is only effective when the type is used for `__key` or `this` fields.
 
-NOTE: To reference another type, you must provide the column list. Otherwise, the column may be automatically resolved as an `OBJECT` type.
+NOTE: `EXTERNAL NAME` aliases are not supported for types, column names must have the same name as their corresponding Java class field or Portable/Compact/Avro schema field.
+
+NOTE: You can mix options that belong to different formats. When you create a mapping that references your UDT, the relevant options are used and the others are ignored, which makes it possible to use a UDT in multiple mappings having different serialization formats.
+
+NOTE: To reference another type, you must provide the column list. Otherwise, in Java format, the column will be resolved as `OBJECT` type; in Portable, Compact and Avro formats, an exception will be thrown once the type is used in a mapping.
 
 === Java Format notes
 For `java` format, if the column list is omitted, it will be automatically resolved from the corresponding
 Java class. Note that the column list will only be extracted from the source class itself;
 if it has columns that are inherited from a superclass, these columns will not be resolved.
 
 === Portable Format notes
-When using the `portable` format, make sure that the `factoryId`, `classId`, and `version` tuples are unique within the cluster.
-Otherwise, deserialization issues can happen if the corresponding class IDs and factory IDs are registered in the client
-for serialization/deserialization.
+When using the `portable` format, make sure that (`factoryId`, `classId`, `version`) tuples are unique within the cluster. Otherwise, deserialization issues may happen.
```

Comment:
You shouldn't change the tuple notation as it has a mathematical meaning, which is very hard to express using plain English: (`factoryId`, `classId`, `version`) tuples. "mut" should be "must".

Reply:
So, this is a mathematical tuple, not a Python tuple data type? A mathematical tuple is a finite sequence of elements, I believe, and the Python data type stores an ordered and unchangeable collection of data.

Are you listing available tuples, or providing an example of a tuple?

Either way, I feel that this needs more explanation, to effectively explain the issue and solution.

And, yes, mut ought to be must ;-) sorry, didn't mean to suggest a new typo!

Reply:
1. "`factoryId`, `classId`, and `version` tuples" means that `factoryId`, `classId`, and `version` are tuples themselves.
2. "(`factoryId`, `classId`, `version`) tuples" means that we are talking about 3-tuples, whose elements denote `factoryId`, `classId` and `version`.

Reply:
Amended the comment so that it, I hope, meets both our concerns

Reply:
"(`factoryId`, `classId`, and `version`)" doesn't conform the tuple notation and "triple" is unnecessary. If there is a way to render LaTeX, you can convert the tuple to a mathematical expression. Otherwise, you have to mix math and plain English. I admit that it might be a bit confusing to the reader, but non-standard notation doesn't help.

Reply:
Alternatively, you can say "that portable IDs (`factoryId`, `classId`, `version`) are unique".